UnityProject/Assets/Scripts/Lighting/LightMountStates.cs

```diff
@@ -241,19 +249,25 @@ private void OnDamageReceived(DamageInfo arg0)
 
 	private void SyncLightState(LightMountState oldState, LightMountState newState)
 	{
-		state = newState;
+		//state = newState;
```

Comment:
you need to set the SyncVar again with the new state in SyncVar hooks. Its silly I know

Reply:
I'm a bit confused by this, would you be able to explain why? In the Mirror docs for hooks it even states that the hook is always called AFTER the property has been set, so wouldn't that mean that it would have already been received and changed by the client before proceeding with the hook?

> The Hook is always called _after_ the property value is set. _You don't need to set it yourself_.

Especially that last part implied that setting it yourself is unnecessary in either case, plus from what I recalled from UNet you should never write to a syncvar variable and instead treat it as read-only due to serialization issues that arise.

Reply:
yep I've read that too, and also tried it in a test project. Client wouldn't take the new value on a hook without setting it. Would be worth another retest now that we have the latest mirror